pkg/compute/rust.go

```diff
@@ -148,6 +170,51 @@ func (r Rust) Verify(out io.Writer) error {
 
 	fmt.Fprintf(out, "Found Cargo.toml at %s\n", fpath)
 
+	// 5) Verify `fastly` crate version
+	//
+	// A valid and up-to-date version of the fastly crate is required.
+	fpath, err = filepath.Abs("Cargo.lock")
+	if err != nil {
+		return fmt.Errorf("error getting Cargo.lock path: %w", err)
+	}
+
+	if !common.FileExists(fpath) {
+		return fmt.Errorf("%s not found", fpath)
+	}
+
+	var lock CargoLock
+	if err := lock.Read("Cargo.lock"); err != nil {
+		return fmt.Errorf("error reading Cargo.lock file: %w", err)
+	}
+
+	var crate CargoPackage
+	for _, p := range lock.Package {
+		if p.Name == "fastly" {
+			crate = p
+		}
+	}
+
+	version, err := semver.Parse(crate.Version)
+	if err != nil {
+		return fmt.Errorf("error parsing Cargo.lock file: %w", err)
+	}
+
+	l, err := GetLatestCrateVersion(r.client, "fastly")
+	if err != nil {
+		return err
+	}
+	latest, err := semver.Parse(l)
+	if err != nil {
+		return err
+	}
+
+	if version.LT(latest) {
+		return errors.RemediationError{
+			Inner:       fmt.Errorf("fastly crate not up-to-date"),
+			Remediation: fmt.Sprintf("To fix this error, run the following command:\n\n\t$ %s\n", text.Bold("cargo update -p fastly")),
```

Comment:
I goofed here. Unfortunately this is not going to be a sufficient remediation when we introduce a semver-breaking change. For example, suppose we have

```toml
[dependencies]
fastly = "0.2.0-alpha4"
```

When I try to use cargo to update it, even specifying the precise version I want, I get:

```
% cargo update -p fastly --precise 0.3.2
    Updating crates.io index
error: failed to select a version for the requirement `fastly = "^0.2.0-alpha4"`
  candidate versions found which didn't match: 0.3.2, 0.3.1, 0.3.0
  location searched: crates.io index
required by package `fastly-template-rust-default v0.1.0 (/home/acfoltzer/src/fastly-template-rust-default)`
```

When I initially proposed using `cargo update`, I forgot that it only updates to versions that are compatible with what's in the `Cargo.toml`. I think the only robust remediation would be to ask them to update that file:

> To fix this error, edit `Cargo.toml` to set `fastly = "^0.3.2"`.

Reply:
Ah yes, that makes sense. I've now updated the message per your suggestion.

<img width="767" alt="Screenshot 2020-05-11 at 19 57 14" src="https://user-images.githubusercontent.com/80089/81600564-b998ee00-93c1-11ea-88fa-2ba6b78cfb9e.png">